File: api/update.py
# api/update.py
import json
from http.server import BaseHTTPRequestHandler
from main import fetch_all

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        try:
            data = fetch_all()
            self.send_response(200)
            self.send_header("Content-Type","application/json")
            self.send_header("Access-Control-Allow-Origin","*")
            self.end_headers()
            self.wfile.write(json.dumps(data, ensure_ascii=False).encode('utf-8'))
        except Exception as e:
            self.send_response(500)
            self.send_header("Content-Type","application/json")
            self.send_header("Access-Control-Allow-Origin","*")
            self.end_headers()
            self.wfile.write(json.dumps({"error": str(e)}).encode('utf-8'))

File: api/update.js
// api/update.js
import fs from "fs";
import path from "path";
import { fetch_all } from "../../main.js"; // 你的 fetch_all 函數

export default async function handler(req, res) {
  try {
    let data = await fetch_all(); // 嘗試抓最新資料
    res.setHeader("Content-Type", "application/json");
    res.setHeader("Access-Control-Allow-Origin", "*");
    res.status(200).json(data);
  } catch (e) {
    console.error("fetch_all 失敗，使用 fallback:", e);

    // fallback 到本地 events.json
    try {
      const filePath = path.join(process.cwd(), "public", "events.json");
      const fileData = fs.readFileSync(filePath, "utf-8");
      const fallbackData = JSON.parse(fileData);

      res.setHeader("Content-Type", "application/json");
      res.setHeader("Access-Control-Allow-Origin", "*");
      res.status(200).json(fallbackData);
    } catch (fileErr) {
      console.error("讀取 fallback 也失敗:", fileErr);
      res.status(500).json({ error: "更新與 fallback 都失敗" });
    }
  }
}